Documents/Automation/EncryptRootVolume/Tests/test_document.py

```diff
@@ -0,0 +1,170 @@
+    retry_count = 12
+    while True:
+        try:
+            sts_client.assume_role(RoleArn=role_arn, RoleSessionName="checking_assume")
+            break
+        except Exception as e:
+            retry_count -= 1
+            if retry_count == 0:
+                raise e
+
+            LOGGER.info("Unable to assume role... trying again in 5 sec")
+            time.sleep(5)
+
+
+class DocumentTest(unittest.TestCase):
+    def test_update_document(self):
+        cfn_client = boto3.client('cloudformation', region_name=REGION)
+        ssm_client = boto3.client('ssm', region_name=REGION)
+
+        ssm_doc = ssm_testing.SSMTester(
+            ssm_client=ssm_client,
+            doc_filename=os.path.join(DOC_DIR,
+                                      'Documents/npark-encryptrootvolume.json'),
+            doc_name=SSM_DOC_NAME,
+            doc_type='Automation'
+        )
+
+        test_cf_stack = ssm_testing.CFNTester(
+            cfn_client=cfn_client,
+            template_filename=os.path.abspath(os.path.join(
+                DOC_DIR,
+                "Tests/CloudFormationTemplates/TestTemplate.yml")),
+            stack_name=TEST_CFN_STACK_NAME
+        )
+
+        LOGGER.info('Creating Test Stack')
+        LOGGER.info('AMI:' + LINUX_AMI_ID)
+        LOGGER.info('Instance Type:' + LINUX_INSTANCE_TYPE)
+        test_cf_stack.create_stack([
+            {
+                'ParameterKey': 'AMI',
+                'ParameterValue': LINUX_AMI_ID
+            },
+            {
+                'ParameterKey': 'INSTANCETYPE',
+                'ParameterValue': LINUX_INSTANCE_TYPE
+            },
+            {
+                'ParameterKey': 'UserARN',
+                'ParameterValue': sts_client.get_caller_identity().get('Arn')
+            }
+        ])
+        LOGGER.info('Test Stack has been created')
+
+        # Verify role exists
+        role_arn = test_cf_stack.stack_outputs['AutomationAssumeRoleARN']
+        verify_role_created(role_arn)
+        try:
+            LOGGER.info("Creating automation document")
+            self.assertEqual(ssm_doc.create_document(), 'Active')
+
+            instance_id = test_cf_stack.stack_outputs['InstanceId']
+            kms_key_id = test_cf_stack.stack_outputs['KmsKeyId']
+
+            execution = ssm_doc.execute_automation(
+                params={'instanceId': [instance_id],
+                        'region': [REGION],
+                        'KmsKeyId': [kms_key_id],
+                        'devicename': ['/dev/xvda'],
+                        'AutomationAssumeRole': [role_arn]})
+            self.assertEqual(ssm_doc.automation_execution_status(ssm_client, execution, False), 'Success')
+
+            LOGGER.info('Encrypting root volume process has been initiated')
+
+            response=ssm_doc.automation_execution_status(ssm_client, execution)
+            if response == 'Success':
```

Comment:
Also assert that the root volume is encrypted as that is the main goal of your Automation.

Reply:
You are not asserting the root volume is encrypted. You'll have to describe the instance root volume within your Python script, get the root volume, describe it, and assert the **Encrypted** property is true:

https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-volumes.html

```json
{
    "Volumes": [
        {
            (...)
            "Encrypted": true,
            (...)
        }
    ]
}
```

Reply:
I've added the assert code.  Please review.